sympy/geometry/polygon.py

```diff
@@ -850,165 +975,23 @@ def _do_poly_distance(self, e2):
-                e2_current, e2_next))
-
-            if (e1_angle < e2_angle) is True:
-                support_line = Line(e1_current, e1_next)
-                e1_segment = Segment(e1_current, e1_next)
-                min_dist_current = e1_segment.distance(e2_current)
-
-                if min_dist_current.evalf() < min_dist.evalf():
-                    min_dist = min_dist_current
-
-                if e1_connections[e1_next][0] != e1_current:
-                    e1_current = e1_next
-                    e1_next = e1_connections[e1_next][0]
-                else:
-                    e1_current = e1_next
-                    e1_next = e1_connections[e1_next][1]
-            elif (e1_angle > e2_angle) is True:
-                support_line = Line(e2_next, e2_current)
-                e2_segment = Segment(e2_current, e2_next)
-                min_dist_current = e2_segment.distance(e1_current)
-
-                if min_dist_current.evalf() < min_dist.evalf():
-                    min_dist = min_dist_current
-
-                if e2_connections[e2_next][0] != e2_current:
-                    e2_current = e2_next
-                    e2_next = e2_connections[e2_next][0]
-                else:
-                    e2_current = e2_next
-                    e2_next = e2_connections[e2_next][1]
-            else:
-                support_line = Line(e1_current, e1_next)
-                e1_segment = Segment(e1_current, e1_next)
-                e2_segment = Segment(e2_current, e2_next)
-                min1 = e1_segment.distance(e2_next)
-                min2 = e2_segment.distance(e1_next)
-
-                min_dist_current = min(min1, min2)
-                if min_dist_current.evalf() < min_dist.evalf():
-                    min_dist = min_dist_current
-
-                if e1_connections[e1_next][0] != e1_current:
-                    e1_current = e1_next
-                    e1_next = e1_connections[e1_next][0]
-                else:
-                    e1_current = e1_next
-                    e1_next = e1_connections[e1_next][1]
-
-                if e2_connections[e2_next][0] != e2_current:
-                    e2_current = e2_next
-                    e2_next = e2_connections[e2_next][0]
-                else:
-                    e2_current = e2_next
-                    e2_next = e2_connections[e2_next][1]
-            if e1_current == e1_ymax and e2_current == e2_ymin:
-                break
+        for _, e1_current, e1_next, _, e2_current, e2_next in _PodalPair(e1, e2, not nested):
+            e1_segment = Segment(e1_current, e1_next)
+            min_dist_current = e1_segment.distance(e2_current)
+            if min_dist_current.evalf() < min_dist.evalf():
+                min_dist = min_dist_current
+
+            e2_segment = Segment(e2_current, e2_next)
+            min_dist_current = e2_segment.distance(e1_current)
+            if min_dist_current.evalf() < min_dist.evalf():
```

Comment:
This can give wrong results if the two answers are near each other. It is better to use `(min_dist_current - min_dist).evalf()`, as that will automatically handle cancellation.

Reply:
That is assuming evalf is really needed here at all. 

Also, are symbolic inputs allowed? If so, this will error.

Reply:
You are right, thanks!

Symbolic inputs are not allowed now, the polygons need to be convex.